tools/src/post-mortem/JSONLog.hs

```diff
@@ -52,7 +53,7 @@ instance Ord IndexedJLTimedEvent where
 runParseLogs :: FilePath -> (Producer IndexedJLTimedEvent IO () -> IO r) -> IO r
 runParseLogs logDir f = do
     xs <- jsonLogs logDir
-    runWithFiles xs ReadMode $ \ys -> f $ interleave (map (uncurry producer) ys)
+    runWithFiles (assert (xs /= []) xs) ReadMode $ \ys -> f $ interleave (map (uncurry producer) ys)
```

Comment:
If you want to ensure you have a non-empty list, you might want to consider `Data.List.NonEmpty` if possible here 😉